core/ipc/rpc/utils/src/main/java/org/opennms/core/rpc/utils/mate/FallbackScope.java

```diff
@@ -48,11 +48,23 @@ public FallbackScope(final Scope... scopes) {
 
     @Override
     public Optional<ScopeValue> get(final ContextKey contextKey) {
-        return this.scopes.stream()
+        Optional<ScopeValue> scopeValue =  this.scopes.stream()
                 .map(scope -> scope.get(contextKey))
                 .filter(Optional::isPresent)
                 .map(Optional::get)
                 .findFirst();
+        // If value itself a Metadata expression, retrieve value for the context key derived from expression.
+        if (scopeValue.isPresent() && Interpolator.containsMateData(scopeValue.get().value)) {
```

Comment:
Should consider opening a backlog tech debt ticket to fix package name and various method names from `mate` -> `meta`, or `mateDate` -> `metaData`, etc.

Reply:
The name was intended to be Mate, see https://www.club-mate.de/en/products/

Reply:
Ahh, thanks for the clarification :)